controllers/provider-aws/pkg/infrastructure/actuator.go

```diff
@@ -0,0 +1,240 @@
+
+func (c *actuator) Update(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return c.reconcile(ctx, config)
+}
+
+func (c *actuator) Delete(ctx context.Context, config *extensionsv1alpha1.Infrastructure) error {
+	return c.delete(ctx, config)
+}
+
+func (c *actuator) reconcile(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	infrastructureConfig := &v1alpha1.InfrastructureConfig{}
+	_, _, err := c.decoder.Decode(infrastructure.Spec.ProviderConfig.Raw, nil, infrastructureConfig)
+	if err != nil {
+		c.injectStatusError(ctx, infrastructure, extensionsv1alpha1.LastOperationTypeReconcile, "could not decode providerConfig", 10, err)
+		return err
+	}
+
+	var (
+		namespace = infrastructure.Spec.SecretRef.Namespace
+		name      = infrastructure.Spec.SecretRef.Name
+	)
+
+	providerSecret := &corev1.Secret{}
+	if err := c.client.Get(ctx, kutil.Key(namespace, name), providerSecret); err != nil {
+		return err
+	}
+	var (
+		createVPC         = true
+		vpcID             = "${aws_vpc.vpc.id}"
+		internetGatewayID = "${aws_internet_gateway.igw.id}"
+		dhcpDomainName    = "ec2.internal"
+		vpcCIDR           = infrastructureConfig.Networks.VPC.CIDR
+		region            = infrastructure.Spec.Region
+	)
+
+	awsClient := aws.NewClient(string(providerSecret.Data[awstypes.AccessKeyID]), string(providerSecret.Data[awstypes.SecretAccessKey]), region)
+
+	// check if we should use an existing VPC or create a new one
+	if len(infrastructureConfig.Networks.VPC.ID) > 0 {
+		createVPC = false
+		vpcID = infrastructureConfig.Networks.VPC.ID
+		igwID, err := awsClient.GetInternetGateway(vpcID)
+		if err != nil {
+			return err
+		}
+		internetGatewayID = igwID
+	} else if len(infrastructureConfig.Networks.VPC.CIDR) > 0 {
+		vpcCIDR = infrastructureConfig.Networks.VPC.CIDR
+	}
+
+	values := map[string]interface{}{
+		"aws": map[string]interface{}{
+			"region": infrastructure.Spec.Region,
+		},
+		"create": map[string]interface{}{
+			"vpc": createVPC,
+		},
+		"sshPublicKey": string(infrastructure.Spec.SSHPublicKey),
+		"vpc": map[string]interface{}{
+			"id":                vpcID,
+			"cidr":              vpcCIDR,
+			"dhcpDomainName":    dhcpDomainName,
+			"internetGatewayID": internetGatewayID,
+		},
+		"clusterName": infrastructure.Spec.SecretRef.Namespace,
+		"zones":       getZones(infrastructure, infrastructureConfig),
+		"outputkeys": map[string]interface{}{
+			"vpcIdKey":                   awstypes.VPCIDKey,
+			"subnetsPublicPrefix":        awstypes.SubnetPublicPrefix,
+			"subnetsNodesPrefix":         awstypes.SubnetNodesPrefix,
+			"securityGroupsNodes":        awstypes.SecurityGroupsNodes,
+			"sshKeyName":                 awstypes.SSHKeyName,
+			"iamInstanceProfileNodes":    awstypes.IAMInstanceProfileNodes,
+			"iamInstanceProfileBastions": awstypes.IAMInstanceProfileBastions,
+			"nodesRole":                  awstypes.NodesRole,
+			"bastionsRole":               awstypes.BastionsRole,
+		},
+	}
+
```

Comment:
https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/awsbotanist/infrastructure.go#L110-L112 <- this code seems to be lost during copy/move.
Can you please try to keep the structure as it [was implemented here](https://github.com/gardener/gardener/blob/master/pkg/operation/cloudbotanist/awsbotanist/infrastructure.go)?